src/compiler/sys.ts

```diff
@@ -557,9 +557,14 @@ namespace ts {
                 fileExists,
                 directoryExists,
                 createDirectory(directoryName: string) {
-                    if (!nodeSystem.directoryExists(directoryName)) {

```

Comment:
Prefer this to be below instead so that we still do not unnecessarily go do fs operation to create directory (many places we cache result of directory exists and would help using that over directly doing to mkdirSync

```typescript
                    if (!nodeSystem.directoryExists(directoryName)) {
                        try {
                            _fs.mkdirSync(directoryName);
                        }
                        catch (e) {
                            if (e.code !== "EEXIST") {
                                throw e;
                            }
                        }
                    }
```

Reply:
Actually on second thought, ignoring "EEXIST" is not right since it could be a file instead of directory so you want this to be:
```typescript
                    if (!nodeSystem.directoryExists(directoryName)) {
                        try {
                            _fs.mkdirSync(directoryName);
                        }
                        catch (e) {
                            if (!nodeSystem.directoryExists(directoryName)) {
                                throw e;
                            }
                        }
                    }
```